src/lib/convert.ts

```diff
@@ -1,30 +1,68 @@
-import Handlebars from 'handlebars';
+import Handlebars, { Exception } from 'handlebars';
+import { String } from 'shelljs';
 
 const convert = input => {
   const transpiler = new Transpiler(input);
   const output = transpiler.toString();
   return output;
 };
 
+class Contexts {
+  private _contexts: any[];
```

Comment:
about the type, I'm not super sure yet. Until now I encountered only two different cases so yes its list of strings but what I'm thinking is we need to store some hints for contexts, maybe a structure like
```
{variables: ['users_i', 'k'], iterateOn: 'array'}
```
or 
```
{variables: ['users_i', 'k'], iterateOn: 'dict'}
```

so that it will be much easier to lookup and say for `@key` if we are iterating over a list we can translate it to `forloop.counter0` and if we are iterating over a dict we can translate it as `{{key}}`

and maybe instead of using a list of dictionaries, we can use a simple class for contexts something like 
```
enum IterateOn {
  list,
  hash
}
class Context {
  variables: string[],
  iterateOn: IterateOn
}
```

what do you think ?

Reply:
I think `variables` will always contain a single string, since we can only introduce 1 new scope variable per context (otherwise we don't know what to prepend it with). Which is correct, since we only introduce the `users_i` value in the new output.

We could also be adding the key (and probably will), but this will only be used to replace the `@key` values, so can be 'hardcoded'.

I like the `iterateOn` idea, but i'm not sure how feasible that is, since we could switch from list to hash if we find a `@key` in a sub-program. Knowing that, I think we should wait with outputting the opening for-statement after we've run outputted the for-body, and retrieve some information. (or do a pre-lookup like I did with the if/else).